### ISSUE TITLE
Negative index error if there are no warps loaded

### ISSUE THREAD

**Issue by ggggg** [BPEssentials/Core#104]
To reproduce:
- have no warps loaded
- do `/warp anything`
- Look at the console, and there would be an error there

Stacktrace:
```
[2022-12-10 19:43:39.313] [ERROR] [EXCEPTION] System.FormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
  at System.Text.StringBuilder.AppendFormatHelper (System.IFormatProvider provider, System.String format, System.ParamsArray args) [0x000f9] in <3b15ef62d6b548c89768f359e89f06c2>:0 
  at System.String.FormatHelper (System.IFormatProvider provider, System.String format, System.ParamsArray args) [0x00023] in <3b15ef62d6b548c89768f359e89f06c2>:0 
  at System.String.Format (System.IFormatProvider provider, System.String format, System.Object arg0, System.Object arg1, System.Object arg2) [0x0000b] in <3b15ef62d6b548c89768f359e89f06c2>:0 
  at BPEssentials.Utils.ChatUtils.FormatMessage (BrokeProtocol.Entities.ShPlayer player, System.String message, System.String formatKey) [0x00069] in <3168d8bba61346d48569fc51a1eb4257>:0 
[2022-12-10 19:44:41.493] [INFO] [COMMAND] The-g: /warp iht
[SVR] Error on packet GlobalMessage: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NullReferenceException: Object reference not set to an instance of an object
  at BPEssentials.Commands.Warp.Invoke (BrokeProtocol.Entities.ShPlayer player, System.String warp) [0x00168] in <3168d8bba61346d48569fc51a1eb4257>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <3b15ef62d6b548c89768f359e89f06c2>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00083] in <3b15ef62d6b548c89768f359e89f06c2>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <3b15ef62d6b548c89768f359e89f06c2>:0 
  at BrokeProtocol.API.Command.Invoke (BrokeProtocol.Entities.ShPlayer player, System.String[] args, System.Int32 inputCount) [0x00043] in <35b15802e53643b6a56104063eb82609>:0 
  at BrokeProtocol.API.CommandHandler.OnEvent (BrokeProtocol.Entities.ShPlayer player, System.String message) [0x000b2] in <35b15802e53643b6a56104063eb82609>:0 
  at BPEssentials.ChatHandlers.GlobalChat.OnEvent (BrokeProtocol.Entities.ShPlayer player, System.String message) [0x0003d] in <3168d8bba61346d48569fc51a1eb4257>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <3b15ef62d6b548c89768f359e89f06c2>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00083] in <3b15ef62d6b548c89768f359e89f06c2>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <3b15ef62d6b548c89768f359e89f06c2>:0 
  at BrokeProtocol.API.DelegateContainer.Execute (System.Object[] args) [0x0008c] in <35b15802e53643b6a56104063eb82609>:0 
  at BrokeProtocol.API.SourceHandler.Exec (System.Int32 eventID, System.Object[] args) [0x0000f] in <35b15802e53643b6a56104063eb82609>:0 
  at BrokeProtocol.API.PlayerEvents.GlobalChatMessage (BrokeProtocol.Entities.ShPlayer player, System.String message) [0x00000] in <35b15802e53643b6a56104063eb82609>:0 
  at BrokeProtocol.Entities.SvPlayer.SvGlobalChatMessage (System.String message) [0x00000] in <35b15802e53643b6a56104063eb82609>:0 
  at BrokeProtocol.Managers.SvManager+<>c.<StartServer>b__99_43 (BrokeProtocol.Entities.ShPlayer sender) [0x00010] in <35b15802e53643b6a56104063eb82609>:0 
  at BrokeProtocol.Managers.SvManager+<ServerLoop>d__129.MoveNext () [0x000d9] in <35b15802e53643b6a56104063eb82609>:0```

**Comment by UserR00T** [comment]
Interesting, what is your warp suggestion mode set to in the config? (`$.Levenshtein.WarpMode`) I have a feeling something is going wrong around those lines, in particular when set to `Suggest` mode. It will fetch the best levenshtein match and display said name, but never null-checks if no matches can be found. [[source]](https://github.com/BPEssentials/Core/blob/feature/v3.1.0/src/BPEssentials/Commands/Warps/Warp.cs#L23-L30) That's my best guess, haven't reproduced it though.

Thanks for reporting the issue and I'll try to reproduce it this weekend!

**Comment by ggggg** [comment]
Hey r00t, I'd normally fix this myself but its exam season, I will check the config after my exam tomorrow!

**Comment by UserR00T** [comment]
Should be fixed in #105, closing!
